### PR TITLE
Upgrade to Rails 6

### PR DESCRIPTION
**Pull request by eoinkelly**
Teach the old dog a new trick :smile:

### CODE REVIEW

**Comment by eoinkelly** [comment]
Fixes #748

**Comment by eoinkelly** [comment]
Codeclimate is reporting a bunch of brakeman issues but I can't replicate it locally. When I run brakeman locally (using brakeman 4.8.2 which is latest version) I get the output I have included at the end of this post.

It seems that Codeclimate can't run Brakeman anymore because its license was changed as part of an aquisition - see

* https://docs.codeclimate.com/docs/brakeman
* https://brakemanscanner.org/blog/2018/06/28/brakeman-has-been-acquired-by-synopsys/index.html

Since we are already running `brakeman` as part of our CI run, I think we can safely disable the outdated version from Codeclimate. I don't see any other way to fix this issue. Sing out if you disagree @Br3nda 

#### Brakeman output when run locally

```
[23:47:13] ❯ be brakeman --run-all-checks --exit-on-warn --format plain
Loading scanner...
Processing application in /Users/eoinkelly/Code/repos/nzsl-online
Processing gems...
[Notice] Detected Rails 6 application
Processing configuration...
[Notice] Escaping HTML by default
Parsing files...
Processing initializers...
Processing libs...ssed
Processing routes...
Processing templates...
Processing data flow in templates...
Processing models...
Processing controllers...
Processing data flow in controllers...
Indexing call sites...
Running checks in parallel...
 - CheckBasicAuth
 - CheckBasicAuthTimingAttack
 - CheckCrossSiteScripting
 - CheckContentTag
 - CheckCookieSerialization
 - CheckCreateWith
 - CheckDefaultRoutes
 - CheckDeserialize
 - CheckDetailedExceptions
 - CheckDigestDoS
 - CheckDynamicFinders
 - CheckEscapeFunction
 - CheckEvaluation
 - CheckExecute
 - CheckFileAccess
 - CheckFileDisclosure
 - CheckFilterSkipping
 - CheckForgerySetting
 - CheckHeaderDoS
 - CheckI18nXSS
 - CheckJRubyXML
 - CheckJSONEncoding
 - CheckJSONEntityEscape
 - CheckJSONParsing
 - CheckLinkTo
 - CheckLinkToHref
 - CheckMailTo
 - CheckMassAssignment
 - CheckMimeTypeDoS
 - CheckModelAttrAccessible
 - CheckModelAttributes
 - CheckModelSerialize
 - CheckNestedAttributes
 - CheckNestedAttributesBypass
 - CheckNumberToCurrency
 - CheckPageCachingCVE
 - CheckPermitAttributes
 - CheckQuoteTableName
 - CheckRedirect
 - CheckRegexDoS
 - CheckRender
 - CheckRenderDoS
 - CheckRenderInline
 - CheckResponseSplitting
 - CheckRouteDoS
 - CheckSafeBufferManipulation
 - CheckSanitizeMethods
 - CheckSelectTag
 - CheckSelectVulnerability
 - CheckSend
 - CheckSendFile
 - CheckSessionManipulation
 - CheckSessionSettings
 - CheckSimpleFormat
 - CheckSingleQuotes
 - CheckSkipBeforeFilter
 - CheckSprocketsPathTraversal
 - CheckSQL
 - CheckSQLCVEs
 - CheckSSLVerify
 - CheckStripTags
 - CheckSymbolDoSCVE
 - CheckTranslateBug
 - CheckUnsafeReflection
 - CheckValidationRegex
 - CheckWithoutProtection
 - CheckXMLDoS
 - CheckYAMLParsing
 - CheckDivideByZero
 - CheckForceSSL
 - CheckReverseTabnabbing
 - CheckSecrets
 - CheckSymbolDoS
 - CheckUnscopedFind
 - CheckWeakHash
Checks finished, collecting results...
Generating report...

== Brakeman Report ==

Application Path: /Users/eoinkelly/Code/repos/nzsl-online
Rails Version: 6.0.0
Brakeman Version: 4.8.2
Scan Date: 2020-05-23 23:47:20 +1200
Duration: 1.680317 seconds
Checks Run: BasicAuth, BasicAuthTimingAttack, ContentTag, CookieSerialization, CreateWith, CrossSiteScripting, DefaultRoutes, Deserialize, DetailedExceptions, DigestDoS, DivideByZero, DynamicFinders, EscapeFunction, Evaluation, Execute, FileAccess, FileDisclosure, FilterSkipping, ForceSSL, ForgerySetting, HeaderDoS, I18nXSS, JRubyXML, JSONEncoding, JSONEntityEscape, JSONParsing, LinkTo, LinkToHref, MailTo, MassAssignment, MimeTypeDoS, ModelAttrAccessible, ModelAttributes, ModelSerialize, NestedAttributes, NestedAttributesBypass, NumberToCurrency, PageCachingCVE, PermitAttributes, QuoteTableName, Redirect, RegexDoS, Render, RenderDoS, RenderInline, ResponseSplitting, ReverseTabnabbing, RouteDoS, SQL, SQLCVEs, SSLVerify, SafeBufferManipulation, SanitizeMethods, Secrets, SelectTag, SelectVulnerability, Send, SendFile, SessionManipulation, SessionSettings, SimpleFormat, SingleQuotes, SkipBeforeFilter, SprocketsPathTraversal, StripTags, SymbolDoS, SymbolDoSCVE, TranslateBug, UnsafeReflection, UnscopedFind, ValidationRegex, WeakHash, WithoutProtection, XMLDoS, YAMLParsing

== Overview ==

Controllers: 13
Models: 15
Templates: 48
Errors: 0
Security Warnings: 0

== Warning Types ==


No warnings found
```

**Comment by Br3nda** [comment]
Nice work! I can give this a test tonight

**Comment by Br3nda** [comment]
I've noticed a small ui change.. 

here's my dev, where you'll see a "change label" button.  but it submits the changed label to the server on every change to the text box (every key typed). It's strange to have a change label button when it's already saved on every key press.
image


Over on production, it doesn't have a "Change label" button

image

**Comment by Br3nda** [comment]
Another small regression (compared to production), going to vocab sheet, then "print', it no longer pops up a print dialogue.

**Comment by Br3nda** [comment]
note: all my testing was in firefox 76.0.1

**Comment by Br3nda** [comment]
search and sign display works great.
admin/cms/passwords all working


just those two small vocab sheet issues. that's all i can find

File: Gemfile [hunk 1]
```diff
@@ -53,11 +62,9 @@ gem 'devise'
 gem 'newrelic_rpm'
 gem 'skylight'
 
-group :assets do
-  gem 'coffee-rails'
-  gem 'sass-rails'
-  gem 'uglifier'
-end
+gem 'coffee-rails'
```
Comment (Br3nda) [inline Gemfile]: we can get rid of this as soon as we fix up `./app/assets/javascripts/ckeditor/config.js.coffee`

Comment (eoinkelly) [reply Gemfile]: Done

**Comment by eoinkelly** [comment]
> I've noticed a small ui change..
> 
> here's my dev, where you'll see a "change label" button. but it submits the changed label to the server on every change to the text box (every key typed). It's strange to have a change label button when it's already saved on every key press.

That bug isn't caused by this PR. Staging has the same issue. I created PR #1007 to revert the change that caused it.

**Comment by eoinkelly** [comment]
@Br3nda I have added ESLint to our CI checks and disabled it in Codeclimate because

1. We are guaranteed to always have a consistent ESLint version
2. I can run the checks locally before pushing which avoids a bunch of conversation with the bot on the PR

**Comment by eoinkelly** [comment]
> Another small regression (compared to production), going to vocab sheet, then "print', it no longer pops up a print dialogue.

Hmmm I can't replicate that locally (tried Firefox 76 and latest Chrome). I created a review app on Heroku and I can replicate it _occasionally_ (like 10% of the time). I don't really know what to do to fix this but I'm adding a 1sec delay before the `window.print()` function is called in the hopes that it will fix this.

**Review by Br3nda — APPROVED** [review]

**Pull request merged**